editor/components/provider/index.js

```diff
@@ -26,33 +25,25 @@ class EditorProvider extends Component {
 
 		// Assume that we don't need to initialize in the case of an error recovery.
 		if ( ! props.recovery ) {
-			this.props.setupEditor( props.post, {
-				...EditorSettings.defaultSettings,
-				...this.props.settings,
-			} );
+			this.props.updateEditorSettings( props.settings );
+			this.props.setupEditor( props.post );
+		}
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( this.props.settings !== prevProps.settings ) {
```

Comment:
I really like this approach, the user changes a preference in the edit post store. That preference changes the props passed to the editor and the editor updates its configuration. I think this is the right way to communicate between stores 👍 
I think we will probably have more uses of this pattern in the future.
The problem we may have is that when settings are updated a rerender happens, then componentDidUpdate is executed, we change the store and we may do another rerender after the store is changed. In this case, I think it is not problematic as we don't depend on the settings of the store, so we don't have double rerenders.
But, I would like to know what are your thoughts on this alternative approach as in other situations the prop may change frequently:
Add a withSelect that selects the current settings from the store, so we have currentSettings (old settings) and settings. Use getDerivedStateFromProps to compare both settings and if they are different call props.updateEditorSettings. In this case, we are updating the store before a re-render but it sounds like a hack. I'm not certain of what would be the correct solution for this pattern.